options.js

```diff
@@ -1,58 +1,128 @@
+function updateColorMenu(colorOptions) {
+  while(page.lastChild) page.removeChild(page.lastChild);
+  buildButtons(colorOptions);
+}
 
-submitButton.onclick = function() {
-  let color = document.getElementById("colorInput").value;
-  let isOk = /^#[0-9A-F]{6}$/i.test(color);
-  const confirmationMessage = document.getElementById("confirmationMessage");
+function colorCodeValidation(color) {
   confirmationMessage.innerText = "";
+  const isOk = /^#([0-9A-F]{3}){1,2}$/i.test(color);
 
   if (!isOk) {
     confirmationMessage.innerText = "Invalid hex code!";
     log("Invalid input");
+    return false;
+  }
+
+  return true;
+}
+
+addButton.onclick = function() {
+  const color = input.value.toUpperCase();
+
+  getColorMenu().then( colorOptions => {
+    if(colorOptions.length === maxMenuSize) {
+      confirmationMessage.innerText = "You have reached the maximum menu size!";
+      log("Max menu size reached");
+      return;
+    }
+
+    if(!colorCodeValidation(color)) return;
+
+    const isInMenu = colorOptions.includes(color);
+
+    if(isInMenu) {
+      confirmationMessage.innerText = "This color is already in the menu";
+      log("Repeated color");
+      return;
+    }
+
+    setExtensionColor(color);
+    updateExtensionColor(color);
+
+    colorOptions.push(color);
+    
+    setColorMenu(colorOptions);
+    updateColorMenu(colorOptions);
+  }); 
+}
+
+removeButton.onclick = function() {
+  const color = input.value.toUpperCase();
+
+  if(!colorCodeValidation(color)) return;
+
+  if(color === crunchyrollOrange) {
+    confirmationMessage.innerText = "You can't remove this color!";
+    log("Tried to remove theme color");
     return;
   }
 
-  setExtensionColor(color);
-  log("Color changed to " + color);
+  getColorMenu().then( colorOptions => {
+    const isInMenu = colorOptions.includes(color);
+
+    if(!isInMenu) {
+      confirmationMessage.innerText = "This color isn't in the menu";
+      log("Color not in menu");
+      return;
+    }
+
+    colorOptions = colorOptions.filter(function(element) { return element != color; });
+
+    getExtensionColor().then( currentColor => {
+      const isInMenu = colorOptions.includes(currentColor);
+      if(!isInMenu) setExtensionColor(crunchyrollOrange);
+    });
+
+    setColorMenu(colorOptions);
+    updateColorMenu(colorOptions);
+  });
+}
+
+function setColorMenu(colorMenu) {
+  chrome.storage.sync.set({ colorOptions: colorMenu }, function () {
+    log("Color menu updated");
+  });
 }
 
 function setExtensionColor(color) {
   chrome.storage.sync.set({ extensionColor: color }, function () {
     log("Setting extension color to " + color);
   });
-  submitButton.style.backgroundColor = color;
 }
 
 input.addEventListener("keyup", function(event) {
   if (event.keyCode === 13) {
     event.preventDefault();
-    submitButton.click();
+    let color = input.value;
+    if(colorCodeValidation(color)) updateExtensionColor(color);
```

Comment:
I think it shouldn't have this behavior, since no button on the page that does that. I think it should just add the color.

Reply:
I've decided to remove the enter behavior